packages/trigger-sdk/src/triggerClient.ts

```diff
@@ -861,6 +862,15 @@ export class TriggerClient {
       },
     };
   }
+
+  defineJob(
```

Comment:
This looks great, but I think `defineJob` needs to be generic so the type of the job returned includes the actual Trigger type (we're eventually going to support direct job invocation, so need to make sure the Trigger type is preserved and not just `any`) Otherwise this whole PR is 👍

Reply:
Done